### PR TITLE
media-gfx/eom: add webp support

### PR DESCRIPTION
**Pull request by oz123**
- [x] I can submit this contribution in agreement with the [Copyright Policy](https://www.gentoo.org/glep/glep-0076.html#certificate-of-origin).
- [x] I have certified the above via adding a `Signed-off-by` line to *every* commit in the pull request.
- [x] This contribution has not been created with the assistance of Natural Language Processing artificial intelligence tools, in accordance with the [AI policy](https://wiki.gentoo.org/wiki/Project:Council/AI_policy).
- [x] I have run `pkgcheck scan --commits --net` to check for issues with my commits.

### CODE REVIEW

**Review by eli-schwartz — CHANGES_REQUESTED** [review]
```diff
diff --git a/media-gfx/eom/eom-1.28.0.ebuild b/media-gfx/eom/eom-1.28.0-r1.ebuild
similarity index 93%
copy from media-gfx/eom/eom-1.28.0.ebuild
copy to media-gfx/eom/eom-1.28.0-r1.ebuild
index 931fc3ee8522..cc82a1b53896 100644
--- a/media-gfx/eom/eom-1.28.0.ebuild
+++ b/media-gfx/eom/eom-1.28.0-r1.ebuild
@@ -7,14 +7,14 @@ inherit mate
 
 MINOR=$(($(ver_cut 2) % 2))
 if [[ ${MINOR} -eq 0 ]]; then
-	KEYWORDS="amd64 ~arm ~arm64 ~loong ~riscv x86"
+	KEYWORDS="~amd64 ~arm ~arm64 ~loong ~riscv ~x86"
 fi
 
 DESCRIPTION="The MATE image viewer"
 LICENSE="FDL-1.1+ GPL-2+ IJG LGPL-2+"
 SLOT="0"
 
-IUSE="X debug exif imagemagick +introspection nls jpeg lcms svg tiff xmp"
+IUSE="X debug exif imagemagick +introspection nls jpeg lcms svg tiff webp xmp"
 
 COMMON_DEPEND="
 	>=app-accessibility/at-spi2-core-2.46.0
@@ -38,6 +38,7 @@ COMMON_DEPEND="
 	jpeg? ( media-libs/libjpeg-turbo:= )
 	lcms? ( media-libs/lcms:2 )
 	svg? ( >=gnome-base/librsvg-2.36.2:2 )
+	webp? ( gui-libs/gdk-pixbuf-loader-webp )
 	xmp? ( >=media-libs/exempi-1.99.5:2= )
 "
```

This seems wrong. It purely adds a new dependency, but doesn't actually add any form of configure switches as a result. According to upstream, webp support is simply automatically handled if you have pixbuf support plus the webp loader is installed.

Have you considered using optfeature.eclass instead, to add a pkg_postinst message that webp support requires installing that package?

**Comment by thesamesam** [comment]
I think I'd made that remark on IRC too.

**Comment by oz123** [comment]
Optfeatue seems like a good idea.
I'll add it, and remove the dependency.

**Review by eli-schwartz — CHANGES_REQUESTED** [review]
New diff:
```diff
diff --git a/media-gfx/eom/eom-1.28.0.ebuild b/media-gfx/eom/eom-1.28.0-r1.ebuild
similarity index 90%
copy from media-gfx/eom/eom-1.28.0.ebuild
copy to media-gfx/eom/eom-1.28.0-r1.ebuild
index 931fc3ee8522..6b4594dfd6c0 100644
--- a/media-gfx/eom/eom-1.28.0.ebuild
+++ b/media-gfx/eom/eom-1.28.0-r1.ebuild
@@ -7,14 +7,14 @@ inherit mate
 
 MINOR=$(($(ver_cut 2) % 2))
 if [[ ${MINOR} -eq 0 ]]; then
-	KEYWORDS="amd64 ~arm ~arm64 ~loong ~riscv x86"
+	KEYWORDS="~amd64 ~arm ~arm64 ~loong ~riscv ~x86"
 fi
 
 DESCRIPTION="The MATE image viewer"
 LICENSE="FDL-1.1+ GPL-2+ IJG LGPL-2+"
 SLOT="0"
 
-IUSE="X debug exif imagemagick +introspection nls jpeg lcms svg tiff xmp"
+IUSE="X debug exif imagemagick +introspection nls jpeg lcms svg tiff webp xmp"
 
 COMMON_DEPEND="
 	>=app-accessibility/at-spi2-core-2.46.0
@@ -70,3 +70,8 @@ src_configure() {
 		$(use_with svg librsvg) \
 		$(use_with xmp)
 }
+
+
+pkg_postinst() {
+	optfeature "For viewing webp images please install" gui-libs/gdk-pixbuf-loader-webp
+}
```

It is still adding in the IUSE for some reason. And underlying that is the fact that the old commit is still there. Please squash the commits together and make it do just one thing.

Note also how the CI is broken:

> MissingInherits	optfeature: missing inherit usage: 'optfeature "For viewing webp images please install" gui-libs/gdk-pixbuf-loader-webp', line 76

**Comment by oz123** [comment]
@eli-schwartz apologies, that was a bit hasty on my side. Thanks for pointing those errors out.

File: media-gfx/eom/eom-1.28.0-r1.ebuild [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+		) \
+		$(use_with jpeg libjpeg) \
+		$(use_with lcms cms) \
+		$(use_with svg librsvg) \
+		$(use_with xmp)
+}
+
+pkg_postinst() {
+	optfeature "For viewing webp images please install" gui-libs/gdk-pixbuf-loader-webp
+}
```
Comment (eli-schwartz) [inline media-gfx/eom/eom-1.28.0-r1.ebuild:76]: This is going to print:

```
 * Install additional packages for optional runtime features:
 *   gui-libs/gdk-pixbuf-loader-webp for For viewing webp images please install
```

**Review by eli-schwartz — CHANGES_REQUESTED** [review]
Aside: I'm not so sure it is true anymore that this is "refreshing the PR XXXX". Also in general I'd recommend pinging immolo in the PR description, not the commit message...

File: media-gfx/eom/eom-1.28.0-r1.ebuild (hunk 1 above)
Comment (immolo) [reply media-gfx/eom/eom-1.28.0-r1.ebuild:76]: @oz123 for future reference you might want to look at the optifeature_header option as show in https://devmanual.gentoo.org/eclass-reference/optfeature.eclass/index.html and https://github.com/gentoo/gentoo/blob/bd9a188b2ca57a74a27ce74686ac1d46b01f2c35/sys-fs/fstransform/fstransform-0.9.4.ebuild#L18

However I'll be happier just "To enable webp support" in this case please.

Comment (oz123) [reply media-gfx/eom/eom-1.28.0-r1.ebuild]: Thanks for reviewing this. I was quite puzzled why the optfeature message was not showing, so it was hard to verify the output.
However, the source code of the eclass is very explicit:

> Print out a message suggesting an optional package (or packages)
> not currently installed which provides the described functionality.

So now I was able to verify that the message is:
```
>>> Installing (1 of 1) media-gfx/eom-1.28.0-r1::oz123-overlay
 * Updating .desktop files database ...                                                                                                                                                                                               [ ok ]
 * Updating icons cache ...                                                                                                                                                                                                           [ ok ]
 * Updating GSettings schemas ...                                                                                                                                                                                                     [ ok ]
 * Install additional packages for optional runtime features:
 *   gui-libs/gdk-pixbuf-loader-webp for webp support
```

File: media-gfx/eom/eom-1.28.0-r1.ebuild [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+		) \
+		$(use_with jpeg libjpeg) \
+		$(use_with lcms cms) \
+		$(use_with svg librsvg) \
+		$(use_with xmp)
+}
+
+pkg_postinst() {
+	optfeature "webp support" gui-libs/gdk-pixbuf-loader-webp
+}
```
Comment (eli-schwartz) [inline media-gfx/eom/eom-1.28.0-r1.ebuild:76]: I was too hasty, and forgot to check this. It overwrote the previous pkg_postinst, which needed to run to handle xdg stuff but now doesn't.

**Pull request closed without merging**